servers/physics_3d/godot_space_3d.cpp

```diff
@@ -175,6 +177,7 @@ bool GodotPhysicsDirectSpaceState3D::intersect_ray(const RayParameters &p_parame
 				min_d = ld;
 				res_point = shape_point;
 				res_normal = inv_xform.basis.xform_inv(shape_normal).normalized();
+				res_face_index = shape->is_concave() ? shape_face_index : -1;
```

Comment:
The `is_concave()` method returns `true` also for `GodotHeightMapShape3D`. The assignment should just be unconditional, and `shape_face_index` should be initialized to `-1`. (Just as all the local `int` variables introduced in this PR should be initialized to `-1`.)

Reply:
Fixed.